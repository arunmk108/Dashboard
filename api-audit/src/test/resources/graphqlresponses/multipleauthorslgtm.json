{
    "data": {
        "repository": {
            "pullRequests": {
                "totalCount": 1,
                "pageInfo": {
                    "endCursor": "Y3Vyc29yOnYyOpK5MjAxOC0wMS0yM1QxNDoxMjo1My0wNTowMM4ADhZl",
                    "hasNextPage": false
                },
                "edges": [
                    {
                        "cursor": "Y3Vyc29yOnYyOpK5MjAxOC0wMS0yM1QxNDoxMjo1My0wNTowMM4ADhZl",
                        "node": {
                            "number": 1,
                            "mergeable": "UNKNOWN",
                            "state": "MERGED",
                            "createdAt": "2018-01-23T19:07:39Z",
                            "updatedAt": "2018-01-23T19:12:53Z",
                            "mergedAt": "2018-01-23T19:12:53Z",
                            "baseRef": {
                                "target": {
                                    "oid": "6375888315f3233e50a56aa6650227dfee7bdd9f"
                                }
                            },
                            "baseRefName": "master",
                            "title": "Merge TestFile into master from dev",
                            "mergeCommit": {
                                "oid": "6375888315f3233e50a56aa6650227dfee7bdd9f"
                            },
                            "headRepository": {
                                "name": "MultipleAuthorsLGTM",
                                "url": "https://mygithub.com/Devopscode/MultipleAuthorsLGTM"
                            },
                            "headRef": {
                                "target": {
                                    "oid": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0"
                                }
                            },
                            "headRefName": "dev",
                            "author": {
                                "login": "ghone",
                                "resourcePath": "/ghone"
                            },
                            "commits": {
                                "totalCount": 3,
                                "nodes": [
                                    {
                                        "commit": {
                                            "oid": "60fef0c4c4b1a0b8dcba7a8d88c54348f75ef5d6",
                                            "committedDate": "2018-01-23T19:06:48Z",
                                            "message": "Create TestFile",
                                            "status": null,
                                            "author": {
                                                "name": "Author, One",
                                                "date": "2018-01-23T14:06:48-05:00",
                                                "user": {
                                                    "login": "ghone"
                                                }
                                            }
                                        }
                                    },
                                    {
                                        "commit": {
                                            "oid": "8331321b9552b2ca3f612b2642c178635d9fa08c",
                                            "committedDate": "2018-01-23T19:07:02Z",
                                            "message": "Update TestFile",
                                            "status": null,
                                            "author": {
                                                "name": "Author, One",
                                                "date": "2018-01-23T14:07:02-05:00",
                                                "user": {
                                                    "login": "ghone"
                                                }
                                            }
                                        }
                                    },
                                    {
                                        "commit": {
                                            "oid": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0",
                                            "committedDate": "2018-01-23T19:09:51Z",
                                            "message": "Create testadd",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2018-01-23T14:09:51-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    }
                                ]
                            },
                            "comments": {
                                "totalCount": 0,
                                "nodes": []
                            },
                            "reviews": {
                                "totalCount": 1,
                                "nodes": [
                                    {
                                        "id": "MDE3OlB1bGxSZXF1ZXN0UmV2aWV3NDk2ODAx",
                                        "bodyText": "LGTM",
                                        "state": "APPROVED",
                                        "author": {
                                            "login": "ghtwo"
                                        },
                                        "createdAt": "2018-01-23T19:12:17Z",
                                        "updatedAt": "2018-01-23T19:12:17Z"
                                    }
                                ]
                            }
                        }
                    }
                ]
            },
            "ref": {
                "target": {
                    "history": {
                        "pageInfo": {
                            "endCursor": "6375888315f3233e50a56aa6650227dfee7bdd9f 4",
                            "hasNextPage": false
                        },
                        "edges": [
                            {
                                "cursor": "6375888315f3233e50a56aa6650227dfee7bdd9f 0",
                                "node": {
                                    "oid": "6375888315f3233e50a56aa6650227dfee7bdd9f",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "6edd4881233c9ed723fc8da84a67502080f266e2"
                                            },
                                            {
                                                "oid": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0"
                                            }
                                        ]
                                    },
                                    "message": "Merge pull request #1 from Devopscode/dev\n\nMerge TestFile into master from dev",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2018-01-23T14:12:52-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2018-01-23T14:12:52-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "6375888315f3233e50a56aa6650227dfee7bdd9f 1",
                                "node": {
                                    "oid": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "8331321b9552b2ca3f612b2642c178635d9fa08c"
                                            }
                                        ]
                                    },
                                    "message": "Create testadd",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2018-01-23T14:09:51-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2018-01-23T14:09:51-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "6375888315f3233e50a56aa6650227dfee7bdd9f 2",
                                "node": {
                                    "oid": "8331321b9552b2ca3f612b2642c178635d9fa08c",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "60fef0c4c4b1a0b8dcba7a8d88c54348f75ef5d6"
                                            }
                                        ]
                                    },
                                    "message": "Update TestFile",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2018-01-23T14:07:02-05:00"
                                    },
                                    "author": {
                                        "name": "Author, One",
                                        "user": {
                                            "login": "ghone"
                                        },
                                        "email": "Author.One@mycompany.com",
                                        "date": "2018-01-23T14:07:02-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "6375888315f3233e50a56aa6650227dfee7bdd9f 3",
                                "node": {
                                    "oid": "60fef0c4c4b1a0b8dcba7a8d88c54348f75ef5d6",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "6edd4881233c9ed723fc8da84a67502080f266e2"
                                            }
                                        ]
                                    },
                                    "message": "Create TestFile",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2018-01-23T14:06:48-05:00"
                                    },
                                    "author": {
                                        "name": "Author, One",
                                        "user": {
                                            "login": "ghone"
                                        },
                                        "email": "Author.One@mycompany.com",
                                        "date": "2018-01-23T14:06:48-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "6375888315f3233e50a56aa6650227dfee7bdd9f 4",
                                "node": {
                                    "oid": "6edd4881233c9ed723fc8da84a67502080f266e2",
                                    "parents": {
                                        "nodes": []
                                    },
                                    "message": "Create README.md",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2018-01-23T14:05:25-05:00"
                                    },
                                    "author": {
                                        "name": "Author, One",
                                        "user": {
                                            "login": "ghone"
                                        },
                                        "email": "Author.One@mycompany.com",
                                        "date": "2018-01-23T14:05:25-05:00"
                                    },
                                    "status": null
                                }
                            }
                        ]
                    }
                }
            },
            "issues": {
                "totalCount": 0,
                "pageInfo": {
                    "endCursor": null,
                    "hasNextPage": false
                },
                "edges": []
            }
        }
    }
}