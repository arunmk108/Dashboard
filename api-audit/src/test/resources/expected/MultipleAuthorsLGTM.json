[
    {
        "auditStatuses": [
            "PEER_REVIEW_LGTM_SUCCESS",
            "COMMITAUTHOR_NE_MERGECOMMITER",
            "PEER_REVIEW_LGTM_SELF_APPROVAL",
            "PULLREQ_NOT_PEER_REVIEWED",
            "GIT_BRANCH_STRATEGY"
        ],
        "lastUpdated": 1516735761179,
        "pullRequest": {
            "scmUrl": "https://mygithub.com/Devopscode/MultipleAuthorsLGTM",
            "scmBranch": "master",
            "scmRevisionNumber": "6375888315f3233e50a56aa6650227dfee7bdd9f",
            "scmCommitLog": "Merge TestFile into master from dev",
            "scmCommitTimestamp": 1516734773000,
            "numberOfChanges": 0,
            "id": "5a678cfcb5d5dc509425d00f",
            "orgName": "Devopscode",
            "repoName": "MultipleAuthorsLGTM",
            "sourceRepo": "Devopscode/MultipleAuthorsLGTM",
            "sourceBranch": "dev",
            "targetRepo": "Devopscode/MultipleAuthorsLGTM",
            "targetBranch": "master",
            "number": "1",
            "collectorItemId": "5a678a032b7af20c95e7b146",
            "updatedAt": 1516734773000,
            "createdAt": 1516734459000,
            "closedAt": 0,
            "state": "merged",
            "mergedAt": 1516734773000,
            "timestamp": 1516734459000,
            "resolutiontime": 0,
            "userId": "ghone",
            "comments": [
                {
                    "user": "ghtwo",
                    "createdAt": 1511190526000,
                    "updatedAt": 1511190526000,
                    "body": "LGTM",
                    "status": ""
                }
            ],
            "reviews": [],
            "commitStatuses": [
                {
                    "state": "SUCCESS",
                    "context": "approvals/lgtmeow",
                    "description": "approved by ghthree"
                }
            ],
            "headSha": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0",
            "baseSha": "6375888315f3233e50a56aa6650227dfee7bdd9f",
            "requestType": "pull",
            "commits": [
                {
                    "scmRevisionNumber": "60fef0c4c4b1a0b8dcba7a8d88c54348f75ef5d6",
                    "scmCommitLog": "Create TestFile",
                    "scmAuthor": "Author, One",
                    "scmAuthorLogin": "ghone",
                    "scmCommitTimestamp": 1516734408000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                },
                {
                    "scmRevisionNumber": "8331321b9552b2ca3f612b2642c178635d9fa08c",
                    "scmCommitLog": "Update TestFile",
                    "scmAuthor": "Author, One",
                    "scmAuthorLogin": "ghone",
                    "scmCommitTimestamp": 1516734422000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                },
                {
                    "scmRevisionNumber": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0",
                    "scmCommitLog": "Create testadd",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmCommitTimestamp": 1516734591000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                }
            ]
        },
        "commits": [
            {
                "scmRevisionNumber": "60fef0c4c4b1a0b8dcba7a8d88c54348f75ef5d6",
                "scmCommitLog": "Create TestFile",
                "scmAuthor": "Author, One",
                "scmAuthorLogin": "ghone",
                "scmCommitTimestamp": 1516734408000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            },
            {
                "scmRevisionNumber": "8331321b9552b2ca3f612b2642c178635d9fa08c",
                "scmCommitLog": "Update TestFile",
                "scmAuthor": "Author, One",
                "scmAuthorLogin": "ghone",
                "scmCommitTimestamp": 1516734422000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            },
            {
                "scmRevisionNumber": "263cabbeb0d00023c3fb6e46c1b9654b56a29ec0",
                "scmCommitLog": "Create testadd",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmCommitTimestamp": 1516734591000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/MultipleAuthorsLGTM",
        "scmBranch": "master"
    },
    {
        "auditStatuses": [
            "DIRECT_COMMITS_TO_BASE_FIRST_COMMIT",
            "SCM_AUTHOR_LOGIN_INVALID"
        ],
        "lastUpdated": 1516735761179,
        "commits": [
            {
                "scmUrl": "https://mygithub.com/Devopscode/MultipleAuthorsLGTM",
                "scmBranch": "master",
                "scmRevisionNumber": "6edd4881233c9ed723fc8da84a67502080f266e2",
                "scmCommitLog": "Create README.md",
                "scmAuthor": "Author, One",
                "scmAuthorLogin": "ghone",
                "scmParentRevisionNumbers": [],
                "scmCommitTimestamp": 1516734325000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a678cfcb5d5dc509425d00e",
                "collectorItemId": "5a678a032b7af20c95e7b146",
                "timestamp": 1516735740805,
                "firstEverCommit": true
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/MultipleAuthorsLGTM",
        "scmBranch": "master"
    }
]